bin/test-changed.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+#
+# Building and testing all the exercises on CI takes a long time
+# and can cause CI to time out.
+# Rather than testing them all we can just test the ones that have
+# changed on the current branch against the master branch.
+#
+
+XPURESCRIPT=$(dirname "$BASH_SOURCE")
+XPURESCRIPT=$(readlink -f "$XPURESCRIPT/..")
+cd "$XPURESCRIPT"
+
+declare -i TEST_RESULT=0
+FAILED_EXERCISES=''
+
+cd exercises
+
+FILES_CHANGED=$(
+  git diff --name-status master..HEAD |
```

Comment:
So in Travis, is this going to always compare against `master`?

Reply:
I was thinking we'd want to compare against whatever revision the commit was based against.

Reply:
In which case you might be able to use something like `HEAD~1`.

Reply:
A PR is always off of master is may not be one commit, so that would be less correct than comparing with master, no?

Reply:
So you want to know which branch the PR was opened against?

Reply:
You could try the `TRAVIS_BRANCH` environment variable.